src/Symfony/Component/Security/Http/Firewall/SimpleFormAuthenticationListener.php

```diff
@@ -107,7 +107,7 @@ protected function attemptAuthentication(Request $request)
             $password = ParameterBagUtils::getRequestParameterValue($request, $this->options['password_parameter']);
         }
 
-        if (!\is_string($username) || (\is_object($username) && !\method_exists($username, '__toString'))) {
+        if (!\is_string($username) && !(\is_object($username) && !\method_exists($username, '__toString'))) {
```

Comment:
Shouldn't it be `!\is_string($username) && (!\is_object($username) || !\method_exists($username, '__toString'))`?

Reply:
You're right, the if statement will fail if it's neither a string nor an object.

I'll fix it.